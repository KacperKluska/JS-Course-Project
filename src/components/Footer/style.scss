footer {
  background-color: #333;
  color: white;
  text-align: center;
  padding: 2rem 3rem 1rem 3rem;

  b {
    margin-bottom: 2rem;
  }
}

.footerCollumns {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-items: flex-start;
  gap: 3rem;

  margin-bottom: 3rem;
}

.footerCollumn {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;

  min-width: 10%;

  label {
    margin-top: 0.3rem;

    &:hover {
      cursor: pointer;
      color: gray;
    }
  }
}
