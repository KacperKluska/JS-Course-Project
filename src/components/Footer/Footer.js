import "./style.scss";

export default function Footer() {
  return (
    <footer>Copyrigth &copy; Kacper Kluska 2021. JavaScript course</footer>
  );
}
