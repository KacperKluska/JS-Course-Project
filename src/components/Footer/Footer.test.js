import Footer from "./Footer";

let wrapper;

beforeAll(() => {
  wrapper = shallow(<Footer />);
});

test("renders", () => {
  expect(wrapper).not.toBeNull();
});

test("footer text", () => {
  wrapper = shallow(<Footer />);
  expect(wrapper.find("footer").text()).toContain(
    `Copyrigth Â© Kacper Kluska 2021. JavaScript course`
  );
});

test("footer text", () => {
  wrapper = shallow(<Footer />);
  expect(wrapper.type()).toBe("footer");
});

test("footer text snapshot", () => {
  const component = renderer.create(<Footer />);
  expect(component.toJSON()).toMatchSnapshot();
});
