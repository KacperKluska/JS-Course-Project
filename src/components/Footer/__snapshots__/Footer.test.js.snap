// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer text snapshot 1`] = `
<footer
  className="footer"
>
  <div
    className="footerCollumns"
  >
    <div
      className="footerCollumn"
    >
      <b>
        <h3>
          Help
        </h3>
      </b>
      <label>
        Returns
      </label>
      <label>
        Change
      </label>
      <label>
        FAQ
      </label>
    </div>
    <div
      className="footerCollumn"
    >
      <b>
        <h3>
          Payment and shipment
        </h3>
      </b>
      <label>
        Payment forms
      </label>
      <label>
        Shipment forms
      </label>
    </div>
    <div
      className="newsletter"
    >
      <b>
        <h3>
          Sign in for our Newsletter
        </h3>
      </b>
      <label
        style={
          Object {
            "textAlign": "start",
          }
        }
      >
        Sign in if you want to be informed about sales and new collections.
      </label>
      <form
        className="newsletterForm"
        onSubmit={[Function]}
      >
        <input
          className="newsletterInput"
          onChange={[Function]}
          placeholder="Your email..."
          required={true}
          type="email"
          value=""
        />
        <button
          className="newsletterButton"
          type="submit"
        >
          Sign in
        </button>
      </form>
    </div>
  </div>
  <div
    className="socialMedias"
  >
    <h3>
      Join us!
    </h3>
    <div>
      <a
        href="https://www.facebook.com/"
        target="_blanc"
      >
        <i
          className="icon-facebook-official"
        />
      </a>
      <a
        href="https://www.youtube.com/"
        target="_blanc"
      >
        <i
          className="icon-youtube"
        />
      </a>
      <a
        href="https://twitter.com/?lang=pl"
        target="_blanc"
      >
        <i
          className="icon-twitter"
        />
      </a>
      <a
        href="https://www.google.com/"
        target="_blanc"
      >
        <i
          className="icon-gplus"
        />
      </a>
    </div>
  </div>
  <b>
    Copyrigth Â© Kacper Kluska 2021. JavaScript course
  </b>
</footer>
`;
